forwardbot_gitlab/models/merge_request.py

```diff
@@ -0,0 +1,276 @@
+
+        pruned_mrs = self.browse([])
+        client = ForwardbotClient(url, token, timeout)
+        for merge_request in merge_requests:
+            if merge_request.target_branch in merge_request.target_project_id.protected_branches:
+                response = client.update_merge_request(
+                    merge_request.target_project_id.remote_id,
+                    merge_request.project_iid,
+                    {"remove_source_branch": False},
+                )
+                _logger.info("Successfully cleared 'remove_source_branch' on %s", merge_request)
+                if response.status_code != 200:
+                    _logger.info("Failed clear 'remove_source_branch' on %s", merge_request)
+
+                if auto_commit:
+                    self.env.cr.commit()  # pylint: disable=invalid-commit
+            else:
+                pruned_mrs |= merge_request
+
+        _logger.info("Pruning merge requests")
+        pruned_mrs.unlink()
+
+    @api.model
+    def process_pending_for_forward_ports(
+        self,
+        auto_commit=False,
+        limit=50,
+        timeout=10,
+        states: Sequence[str] = None,
+    ):
+        url, token = get_api_data(self.env)
+        if not url or not token:
+            return
+
+        if states is None:
+            states = ["merged"]
+
+        pending_for_fw: MergeRequest = self.search(
+            [
+                ("processed", "=", False),
+                ("state", "in", states),
+            ],
+            limit=limit,
+            order="id",
+        )
+        _logger.info("Will process %d MRs waiting for a forward port", len(pending_for_fw))
+        if not pending_for_fw:
+            return
+
+        client = ForwardbotClient(url, token, timeout)
+        for merge_request in pending_for_fw:
+            try:
+                merge_request.create_forward_port()
+                forward_port = merge_request.forward_port_id
+                if not forward_port:
+                    _logger.info(
+                        "No stable branch after %s for project %d. Forward port chain stopped",
+                        merge_request.target_branch,
+                        merge_request.target_project_id.remote_id,
+                    )
+                    response = client.delete_branch(merge_request.source_project_id, merge_request.source_branch)
+                    if response.status_code not in [204, 404]:
+                        _logger.warning("Received status code %d while deleting source branch %s")
+
+                    merge_request.processed = True
+                    continue
+
+                _logger.info("Forward port %s created", forward_port)
+
+                # Clone branch used for forward port
+                response = client.create_branch(
+                    forward_port.source_project_id, forward_port.source_branch, merge_request.source_branch
+                )
+                if response.status_code == 400:
```

Comment:
IMHO we should use response.raise_for_status and process the result of the message of the API response